<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TableViewWidget</class>
 <widget class="QWidget" name="TableViewWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>926</width>
    <height>682</height>
   </rect>
  </property>
  <property name="styleSheet">
   <string notr="true">#TableViewWidget{
background-color:white;
}
QAbstractItemView {
background-color: white;
color: black;
border: 0px;
}
QPushButton, QComboBox {
border-image: url(:/tulip/gui/ui/btn_26.png) 4;
border-width: 4;
padding: 0px 6px;
font-size: 12px;
}

QPushButton::flat {
border-image: none;
border-width: 4;
padding: 0px 6px;
font-size: 12px;
background-color: transparent;
}

QPushButton, QComboBox, QLabel {
color: black;
}

QPushButton:hover, QComboBox:hover {
border-image: url(:/tulip/gui/ui/btn_26_hover.png) 4;
}

QPushButton:disabled, QComboBox::disabled {
color:gray;
}

QPushButton:pressed{
border-image: url(:/tulip/gui/ui/btn_26_pressed.png) 4;
}

QPushButton::menu-indicator{
subcontrol-origin: margin;
subcontrol-position: center right;
right: 4px;
}

QPushButton {
outline: none;
margin: 2
}

QComboBox::down-arrow {
image: url(:/tulip/gui/ui/combobox_arrow.png);
}

QComboBox:drop-down {
subcontrol-origin: padding;
subcontrol-position: top right;
border-left-style: none; 
border-top-right-radius: 1px;
border-bottom-right-radius: 1px;
}

QDockWidget:title {
border-image:none;
background-color: qlineargradient(x1: 0, y1: 0, x2: 0.0, y2: 1.0,
stop: 0 #838383,
stop: 0.4 #707070,
stop: 0.401 #636363,
stop: 1 #4a4a4a);
border: 0px solid black;
color: white;
text-align: center;
color: white;
}

QSplitter::handle {
background-color: #C0C0C0;
border-width:0px;
}

QFrame[ bottomFrame = &quot;true&quot; ] * {
color: black;
font: bold 11px;
}

QFrame[ bottomFrame = &quot;true&quot; ] {
border-top: 1px solid rgba(117,117,117,255);
border-bottom: 1px solid rgba(117,117,117,255);
border-right: 1px solid rgba(117,117,117,255);
border-left: 0px;
background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:, y2:1, 
stop:0 rgb(75,75,75),
stop:1 rgb(60, 60, 60));
}

QFrame[ bottomFrame = &quot;true&quot; ] QPushButton {
border: 0px;
border-image: none;
}

QFrame[ bottomFrame = &quot;true&quot; ] QPushButton:hover {
border: 0px;
border-image: none;
background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:, y2:1, 
stop:0 rgb(85,85,85),
stop:1 rgb(70, 70, 70));
}

QFrame[ bottomFrame = &quot;true&quot; ] QPushButton:pressed, QFrame[ bottomFrame = &quot;true&quot; ] .QPushButton:checked {
border: 0px;
border-image: none;
background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:, y2:1, 
stop:0 rgb(105,105,105),
stop:1 rgb(90, 90, 90));
}
</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="frame">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QFrame" name="frame_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>42</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true">font-weight: bold;
color: #595959;</string>
           </property>
           <property name="text">
            <string>&lt;font size=&quot;5&quot;&gt;Elements&lt;/font&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="spacing">
            <number>5</number>
           </property>
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Show</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="eltTypeCombo">
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContents</enum>
             </property>
             <property name="frame">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="filteringLabel">
             <property name="text">
              <string>filtered by</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="filteringPropertyCombo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContents</enum>
             </property>
             <property name="frame">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Expanding</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="frame_3">
        <property name="styleSheet">
         <string notr="true">#frame_3 {
padding-top: 5px;
}</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="spacing">
            <number>5</number>
           </property>
           <property name="leftMargin">
            <number>5</number>
           </property>
           <property name="bottomMargin">
            <number>5</number>
           </property>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>matching</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="ClearableLineEdit" name="filterEdit">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>120</width>
               <height>24</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true">background-color: white;
border: 0px;
color: black;
border-bottom: 1px solid #C9C9C9;
border-top: 1px solid #C9C9C9;
border-left: 1px solid #C9C9C9;</string>
             </property>
             <property name="placeholderText">
              <string>give a pattern</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>in</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="matchPropertyButton">
             <property name="layoutDirection">
              <enum>Qt::RightToLeft</enum>
             </property>
             <property name="text">
              <string>Any</string>
             </property>
             <property name="icon">
              <iconset resource="../../../library/tulip-gui/resources/TulipGUIResource.qrc">
               <normaloff>:/tulip/gui/ui/combobox_arrow.png</normaloff>:/tulip/gui/ui/combobox_arrow.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>column</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="NavigableTableView" name="table">
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <property name="sortingEnabled">
            <bool>true</bool>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
           <property name="cornerButtonEnabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ClearableLineEdit</class>
   <extends>QLineEdit</extends>
   <header location="global">tulip/ClearableLineEdit.h</header>
  </customwidget>
  <customwidget>
   <class>NavigableTableView</class>
   <extends>QTableView</extends>
   <header>navigabletableview.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../library/tulip-gui/resources/TulipGUIResource.qrc"/>
 </resources>
 <connections/>
 <slots>
  <slot>setPropertyVisible(tlp::PropertyInterface*,bool)</slot>
  <slot>setFilter(QString)</slot>
 </slots>
</ui>
